### PR TITLE
CCM

### PR DESCRIPTION
**Pull request by sjpadgett**
- deprecated create_function replace w/anonymous function
- focus ccm import and review.
- some work on immunizations.

#### Short description of what this resolves:
- CCDA import, review, view CCDA and patient add/merge are working again except a couple workflow issues which i'll address later with an interface PR. Still need to fix so a complete ccda import is parsed.
- Partial immunization module fixes. still some maria/sql hiccups and a stubborn utility class to deal with.
- HIE fixed to queue up CCDA for dispatch

### CODE REVIEW

**Comment by sjpadgett** [comment]
I'll be bringing this in as a testing base for demo reset later. Further PR's will be by module for ease of tracking and my sanity.
This fixes many of the shared resources across all modules so...
Hopefully Travis is my friend again!:)

File: interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php [hunk 1]
```diff
@@ -1969,21 +1969,16 @@ public function logCCDA($pid, $encounter, $content, $time, $status, $user_id, $v
...
-            $docid = $resp->id;
-            $revid = $resp->rev;
+            $data = array(
+                'data'      => base64_encode($content),
+                'pid'       => $pid,
+                'encounter' => $encounter,
+                'mimetype'  => 'text/xml'
+            );
+            $couch    = \Documents\Plugin\Documents::couchDB();
+            $couch_id = \Documents\Plugin\Documents::saveCouchDocument($couch, $data);
```
Comment (bradymiller) [inline interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php:1981]: looks like recent code is getting reverted

Comment (bradymiller) [reply interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php:1981]: looks like just in this script and in the below EncountermanagerTable.php script

Comment (bradymiller) [reply interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php:1981]: i can sort this out quickly after in codebase (if wanna keep the momentum going :) )

**Comment by bradymiller** [comment]
the travis reaper strikes :)

tenor (41)

File: interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php (hunk 1 above)
Comment (sjpadgett) [reply interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php:1981]: thanks Eagle Eye but, i'll go ahead and fix so I don't have to pull in again as I have many current changes.
Who comes up with some of this PSR nonsense!! PHP has no problem with compounding name spaces as many reasonable levels as I wish.... I hate arbitrary junk like this. Or maybe I just hate being caught at it!:)

**Comment by sjpadgett** [comment]
looks good to me Brady.
I still don't understand why authors would instantiate some main classes in one sub module and run static in another! I still need to deal with using static db adapters as I think at some point they will be problematic.

**Comment by bradymiller** [comment]
looks good to me. the whole static thing in the ccm module I think was seeded initially when php didn't throw warning/errors about it and then turned into more of a windy road avoiding php warnings/errors as newer php versions correctly began pointing out these issues. I'm just glad your doing this and not me :)

**Comment by sjpadgett** [comment]
more likely two different developers not being in sync. teamwork folks!:)
thanks! bringing in then get immunizations working then some UI stuff.

**Pull request merged**